services/catalog/service/migrations/committed/000011-localizations-added.sql

```diff
@@ -1,6 +1,107 @@
+  the underlying table. Naming conflicts are avoided using Postgraphile smart
+  comments.
+  It is recommended to use smart tags to fine-tune the resulting GraphQL schema,
+  such as adding virtual constraints, adding comments, and marking fields as not-null.
+*/
+CREATE OR REPLACE FUNCTION app_private.define_localization_view(localizableFields text[], tableName text, localizationsTableName text, fkColumn text) RETURNS void
+    LANGUAGE plpgsql
+    AS $$
+DECLARE
+  localizableFieldsSelect text = 'l.' || array_to_string(localizableFields, ', l.');
+BEGIN
+  EXECUTE 'DROP VIEW IF EXISTS app_public.' || tableName || '_view CASCADE;';
+  EXECUTE 'CREATE VIEW app_public.' || tableName || '_view AS ' ||
+          'SELECT p.*, c.* FROM app_public.' || tableName || ' p ' ||
+          'LEFT OUTER JOIN LATERAL ( ' ||
+          'SELECT ' || localizableFieldsSelect || ' ' ||
+          'FROM app_public.' || localizationsTableName || ' l ' ||
+          'WHERE l.' || fkColumn || ' = p.id AND (l.locale = (SELECT pg_catalog.current_setting('':MOSAIC_LOCALE'', true)) OR l.is_default_locale IS TRUE) ' ||
+          'ORDER BY l.is_default_locale ASC LIMIT 1) c ON TRUE;';
+
+  EXECUTE 'GRANT SELECT ON app_public.' || tableName || '_view TO ":DATABASE_GQL_ROLE";';
+
+  EXECUTE 'COMMENT ON TABLE app_public.' || tableName || ' IS E''@omit\n@name ' || tableName || '_data'';';
+  EXECUTE 'COMMENT ON TABLE app_public.' || localizationsTableName || ' IS E''@omit'';';
+  EXECUTE 'COMMENT ON VIEW app_public.' || tableName || '_view IS E''@name ' || tableName || '\n@primaryKey id'';';
+END;
+$$;
+
+-- Movies
+CREATE TABLE IF NOT EXISTS app_public.movie_localizations(
+  id INTEGER PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+  movie_id TEXT REFERENCES movie ON DELETE CASCADE,
+  locale TEXT NOT NULL,
+  is_default_locale BOOLEAN NOT NULL,
+  title TEXT NOT NULL,
+  description TEXT,
+  synopsis TEXT
+);
+
+SELECT ax_define.define_index('movie_id', 'movie_localizations', 'app_public');
+SELECT ax_define.define_index('locale', 'movie_localizations', 'app_public');
+
+-- Migrate values for the default locale into the table.
+DO $$ BEGIN
+  IF ax_define.column_exists('title', 'movie', 'app_public') THEN
+    INSERT INTO app_public.movie_localizations
+    (movie_id, title, description, synopsis, locale, is_default_locale)
+    SELECT m.id, m.title, m.description, m.synopsis, ':DEFAULT_LOCALE_TAG', true
+    FROM app_public.movie m
+    ON CONFLICT DO NOTHING;
+  END IF;
+END $$;
+
+ALTER TABLE app_public.movie DROP COLUMN IF EXISTS title;
+ALTER TABLE app_public.movie DROP COLUMN IF EXISTS description;
+ALTER TABLE app_public.movie DROP COLUMN IF EXISTS synopsis;
+
+SELECT app_private.define_localization_view(
+  ARRAY['title', 'description', 'synopsis'],
+  'movie',
+  'movie_localizations',
+  'movie_id');
+
+GRANT SELECT, INSERT, UPDATE, DELETE ON app_public.movie_localizations TO ":DATABASE_GQL_ROLE";
```

Comment:
The GQL role should not need insert/update/delete as this is done from message handlers which should use the owner role. Also for the other tables below.

Reply:
Message handlers are currently using the GQL role. We should change this as part of another PR.